src/solvers/smt2/smt2_parser.cpp

```
@@ -864,6 +962,33 @@ exprt smt2_parsert::expression()
         return true_exprt();
       else if(identifier==ID_false)
         return false_exprt();
+      else if(identifier == "roundNearestTiesToEven")
+      {
+        // we encode as 32-bit unsignedbv
+        return from_integer(ieee_floatt::ROUND_TO_EVEN, unsignedbv_typet(32));
+      }
+      else if(identifier == "roundNearestTiesToAway")
+      {
+        // we encode as 32-bit unsignedbv
+        return from_integer(4, unsignedbv_typet(32)); // we don't have it
```

How does the solver behave in this case? If it crashes or returns wrong results then it would better to throw an error here.

---

done